src/Microsoft.DotNet.Build.Tasks/ExecWithRetriesForNuGetPush.cs

```diff
@@ -0,0 +1,125 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Tasks;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    /// <summary>
+    /// Run a command and retry if the exit code is not 0.
+    /// </summary>
+    public class ExecWithRetriesForNuGetPush : ExecWithRetries
+    {
+        /// <summary>
+        /// There's a very special failure scenario that we want to ignore.  That scenario is
+        /// when NuGet hits a timeout on one "push" attempt, and then gets a "Forbidden" response
+        /// because the package "already exists" on the next response.  This indicates that the
+        /// timeout occurred, but the push was actually successful.
+        ///
+        /// To address the condition, I'm implementing a special "IgnoredErrorMessagesWithConditional"
+        /// property that allows you to specify error messages which you want to ignore.  If you
```

Comment:
`property` -> `item collection` (msbuild perspective, and clarifies reference to Item later).